compiler/damlc/daml-stdlib-src/DA/List.daml

```diff
@@ -255,6 +256,22 @@ stripPrefix _ _ = None
 stripSuffix : Eq a => [a] -> [a] -> Optional [a]
 stripSuffix a b = reverse <$> stripPrefix (reverse a) (reverse b)
 
+-- | Return the string before and after the search string or `None`
+-- if the search string is not found.
+--
+-- ```
+-- >>> stripInfix [0,0] [1,0,0,2,0,0,3]
+-- Some ([1], [2,0,0,3])
+--
+-- >>> stripInfix [0,0] [1,2,0,4,5]
+-- None
+-- ```
+stripInfix : Eq a => [a] -> [a] -> Optional ([a], [a])
```

Comment:
Bike shedding alarm: should this be called something like `splitInfix`?

Reply:
I prefer `stripInfix` since it matches `stripPrefix` and `stripSuffix` (and that’s also what the `extra` lib in Haskell land calls this).

Reply:
Sounds reasonable. Let's go with `stripInfix` then.